### PR TITLE
remove schema_cache.dump and all references to it

### PR DESCRIPTION
**Pull request by davidsbailey**
## Background

Rails 5.1 moved the schema cache file from `schema_cache.dump` to `schema_cache.yml` ([source](https://kirshatrov.com/posts/schema-cache/#:~:text=Implementation)). Our system has mostly moved over to using the new file, but the old file is still getting generated by the staging machine and is still referenced by several parts of our code.

Timeline:
* Oct 2020 - we upgraded to rails 5.1: https://github.com/code-dot-org/code-dot-org/commit/8120eb156d9f3917a47b97af7ad6a345d05b7fe9
* Jan 2021 -  I made a narrowly scoped change to make staging commit changes to the new file instead of the old file

Today I noticed that schema_cache.dump is still getting generated on the staging machine and I wasted a bit of time debugging it. my conclusion was that this file is not actually hurting anything in our automated system, but is a nuisance to developers trying to debug schema cache problems, therefore it would be best to remove it. some additional observations on this can be seen in https://github.com/code-dot-org/code-dot-org/pull/41622 . 

## Description

This PR removes all references to schema_cache.dump from our codebase and adds it to the .gitignore file, so that we do not accidentally commit it if it appears again. 

## Caveats

this PR also removes two chunks of code which comments suggest are dedicated to normalizing the binary format of the old schema_cache.dump file. I don't have a clear sense of whether https://github.com/code-dot-org/code-dot-org/pull/41626/commits/6f9eace72efda2dd26f99cf92e742195e5980cfe in particular is still necessary given the new .yml format. I could revert that commit for now if reviewers think that's safer.

## Deployment strategy

I am planning to just merge this PR and see if anything breaks, then revert if necessary.

### CODE REVIEW

File: dashboard/.gitignore [hunk 1]
```diff
@@ -24,3 +24,5 @@
 
 /public/code-studio-package
 /public/apps-package
+
+/db/schema_cache.dump
```
Comment (wjordan) [inline dashboard/.gitignore]: I don't think this should be in `.gitignore`, we don't want/need this file to exist anywhere in any environment unless I'm mistaken?

**Review by wjordan — APPROVED** [review]
Generally looks good to me. See [FND-1521](https://codedotorg.atlassian.net/browse/FND-1521) for reference to the tracked item.

**Review by wjordan — APPROVED** [review]

**Comment by sureshc** [comment]
Thank you for cleaning this up! We have a Jira Task (in Foundations?) for this, which I will find and reference here. A related work item, I think is to ensure that our test and production environments are configured to use schema_cache.yml file (and NOT query the database schema). I did some research on these settings and will find the Jira Task with my notes for that.

**Comment by sureshc** [comment]
Do we need to update our Rails config files? It appears [the setting to use the schema cache file](https://edgeguides.rubyonrails.org/configuring.html#configuring-active-record) may have changed from `config.use_schema_cache.dump` to `config.active_record.use_schema_cache.dump`. I assume (hope) that Rails 5.2 is backwards compatible with the older configuration key.

```
[production] dashboard > Rails.application.config.active_record.use_schema_cache_dump
=> nil
[production] dashboard > Rails.application.config.use_schema_cache_dump
=> true
```

On staging, should we explicitly NOT use the dump file since this environment is responsible for querying the database schema, generating a schema cache file, and committing the authoritative cache file? [Currently staging is configured](https://github.com/code-dot-org/code-dot-org/blob/b2114c456f3faa73967604e11ee0db7a7db895e1/dashboard/config/environments/staging.rb#L20) to use the cache file.

**Comment by davidsbailey** [comment]
Good questions @sureshc . a few responses:

First of all, I'm assuming this is all a discussion of potential future work. if there's anything in here which should block this PR, please clarify.

> On staging, should we explicitly NOT use the dump file since this environment is responsible for querying the database schema, generating a schema cache file, and committing the authoritative cache file?

I think the answer is no, because that's not the order things happen in. my understanding is that staging updates schema_cache.yml during database migration, not during application startup, and that use_schema_cache_dump only affects application startup. in case my theory is wrong, in practice it appears that staging has been generating and committing updates to schema_cache.yml, and (in theory) this PR shouldn't change that.

> [Currently staging is configured](https://github.com/code-dot-org/code-dot-org/blob/b2114c456f3faa73967604e11ee0db7a7db895e1/dashboard/config/environments/staging.rb#L20) to use the cache file.

Interestingly, ALL environments are currently configured to use the schema cache dump, because `config.active_record.use_schema_cache_dump` Defaults to `true`. (see https://guides.rubyonrails.org/v5.2/configuring.html#configuring-active-record). 

My conclusion is to remove the useless `config.use_schema_cache_dump = true` statements for all environments, since they are both ~redundant~ using the old param name and also trying to set something that is already the default.

**Review by Hamms — APPROVED** [review]
Looks great to me! Thanks for taking this on

**Pull request merged**